iceoryx_hoofs/source/posix_wrapper/thread.cpp

```diff
@@ -49,5 +50,111 @@ ThreadName_t getThreadName(pthread_t thread) noexcept
     return ThreadName_t(cxx::TruncateToCapacity, tempName);
 }
 
+cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread,
+                                                 const Thread::callable_t& callable) noexcept
+{
+    if (!callable)
```

Comment:
Not part of this PR but we really should get rid of nullable callables.

Reply:
That is debatable and I disagree for the purpose of storage (what `std::function` and `cxx::function` is about). The issue here is that it is used for passing callable, which is not its primary purpose. This could be avoided with a template function as for `std::thread` which will make it impossible to pass anything that is not callable.

This requires distinguishing some cases in generic programming but is the way to go IMO.